scripts/pre_commit_linter.py

```diff
@@ -180,6 +182,16 @@
 _MESSAGE_TYPE_FAILED = 'FAILED'
 
 
+def _is_filename_excluded_for_bad_patterns_check(filename):
+    """Checks if file is excluded from the bad patterns check
+
+    Returns:
+        a boolean: true if excluded
+    """
+    return (any(filename.startswith(bad_pattern)
+                for bad_pattern in BAD_PATTERNS['\t']['excluded_dirs'])
```

Comment:
Actually, there's an important principle here that is important to keep in mind. While fixing an issue, don't break existing functionality. Test to make sure everything is still covered, and be extra careful when refactoring existing code.